vscode-trace-webviews/src/trace-viewer/vscode-trace-viewer-container.tsx

```diff
@@ -163,10 +193,162 @@ class TraceViewerContainer extends React.Component<{}, VscodeAppState>  {
 
   protected async doHandleExperimentSetSignal(experiment: Experiment| undefined): Promise<void> {
       if (experiment) {
+          if (this.state.experiment === undefined) {
+              await this.fetchMarkerSets(experiment.UUID);
+          }
           const defaultOverviewDescriptor: OutputDescriptor | undefined  = await this.getDefaultTraceOverviewOutputDescriptor(experiment);
           this.setState({
               experiment: experiment,
               overviewOutputDescriptor: defaultOverviewDescriptor});
+          this._signalHandler.setMarkerSetsContext(this.markerSetsMap.size > 0);
+          this._signalHandler.setMarkerCategoriesContext(this.toolbarMarkerCategoriesMap.size > 0);
+      }
+  }
+
+  protected doHandleExperimentSelectedSignal(experiment: Experiment| undefined): void {
+      if (experiment?.UUID === this.state.experiment?.UUID) {
+          this._signalHandler.setMarkerSetsContext(this.markerSetsMap.size > 0);
+          this._signalHandler.setMarkerCategoriesContext(this.toolbarMarkerCategoriesMap.size > 0);
+      }
+  }
+
+  protected async doHandleOutputAddedMessage(descriptor: OutputDescriptor): Promise<void> {
+      if (!this.state.outputs.find(output => output.id === descriptor.id)) {
+          this.setState({outputs: [...this.state.outputs, descriptor] });
+          await this.fetchAnnotationCategories(descriptor);
+      }
+  }
+
+  private async fetchMarkerSets(expUUID: string): Promise<void> {
+      if (this.state.tspClientProvider) {
```

Comment:
@hriday-panchasara I was testing it and initially it was working. But then for some unknown reason the hamburger menu doesn't show-up anymore. I set a breakpoint here and `this.state.tspClientProvider` is undefined. I think that's the reason. Anything else seems to work. Any ideas what goes wrong?

Reply:
adding `this.fetchMarkerSets(message.experiment.UUID);` below line 90 makes it work. I'm not sure if it is the right way and you know better.

Reply:
In some cases `SET_EXPERIMENT` is caught before `SET_TSP_CLIENT`, causing `fetchMarkerSets` to be skipped. Let me know if it works now